Library/Homebrew/cmd/formulae.sh

```diff
@@ -8,5 +8,17 @@
 source "${HOMEBREW_LIBRARY}/Homebrew/items.sh"
 
 homebrew-formulae() {
-  homebrew-items '*\.rb' 'Casks' 's|/Formula/|/|' '^homebrew/core'
+  local formulae
+  formulae="$(homebrew-items '*\.rb' 'Casks' 's|/Formula/|/|' '^homebrew/core')"
+
+  # HOMEBREW_CACHE is set by brew.sh
+  # shellcheck disable=SC2154
+  if [[ -n "${HOMEBREW_INSTALL_FROM_API}" && -f "${HOMEBREW_CACHE}/api/formula.json" ]]
+  then
+    local api_formulae
+    api_formulae="$(ruby -e "require 'json'; JSON.parse(File.read('${HOMEBREW_CACHE}/api/formula.json')).each { |f| puts f['name'] }")"
+    formulae=$(echo -e "${formulae}\n${api_formulae}" | sort -uf | grep .)
```

Comment:
Concept idea for reading cached JSON using ruby without brew libraries.

This currently doesn't handle any error inside Ruby. Particularly, a corrupt JSON file ends up dumping the entire contents of formula.json to terminal which can be quite lengthy.

I had added `grep .` since `${formulae}` can be `""` so it leads to blank line. Not sure if there is better way to handle this.